winch/codegen/src/isa/aarch64/asm.rs

```diff
@@ -473,6 +475,67 @@ impl Assembler {
         );
     }
 
+    /// Signed/unsigned remainder operation with three registers.
+    pub fn rem_rrr(
+        &mut self,
+        divisor: Reg,
+        dividend: Reg,
+        dest: Writable<Reg>,
+        kind: RemKind,
+        size: OperandSize,
+    ) {
+        // Check for division by 0
+        self.emit(Inst::TrapIf {
+            kind: CondBrKind::Zero(divisor.into()),
+            trap_code: TrapCode::INTEGER_DIVISION_BY_ZERO,
+        });
+
+        // `cranelift-codegen` doesn't support emitting u/sdiv for anything but I64,
+        // we therefore sign-extend the operand.
+        // see: https://github.com/bytecodealliance/wasmtime/issues/9766
+        //
+        // FIXME: maybe this is only necessary for signed remainder?
```

Comment:
@saulecabrera I'm not sure about that, but if we are performing an unsigned remainder operation, then do we really care about sign extending? Maybe it's not necessary for division either? Is there a scenario where we can't just blindly consider the `w` register as a `x` register?

Reply:
The same reasonning applies to div, I think. We could shave a few instructions in the case of unsigned division too.